tools/src/main/java/org/apache/kafka/tools/CommandOptions.java

```diff
@@ -0,0 +1,80 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements. See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License. You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.kafka.tools;
+
+import net.sourceforge.argparse4j.ArgumentParsers;
+import net.sourceforge.argparse4j.inf.ArgumentParser;
+import net.sourceforge.argparse4j.inf.ArgumentParserException;
+import net.sourceforge.argparse4j.inf.Namespace;
+
+/**
+ * Base class for command line tools options
+ */
+public abstract class CommandOptions {
+
+    protected final ArgumentParser parser;
+    protected final Namespace ns;
+
+    /**
+     * Constructor
+     *
+     * @param command   command
+     * @param description   command description
+     * @param args  arguments to parse for building command options
+     */
+    public CommandOptions(String command, String description, String[] args) {
+
+        this.parser = ArgumentParsers
+                .newArgumentParser(command)
+                .defaultHelp(true)
+                .description(description);
+
+        this.prepareArgs();
+
+        if (args.length == 0) {
+            CommandLineUtils.printUsageAndDie(this.parser, description);
+        }
+
+        Namespace ns = null;
+        try {
+            ns = this.parser.parseArgs(args);
+        } catch (ArgumentParserException e) {
+            CommandLineUtils.printErrorAndDie(this.parser, e);
+        }
+        this.ns = ns;
+    }
+
+    /**
+     * If an option was specified on the command line
+     *
+     * @param option    option to check
+     * @return  if the option was specified
+     */
+    public boolean has(String option) {
+
+        if (this.ns.get(option) instanceof Boolean)
```

Comment:
why is it so specific for Boolean type?

Reply:
@adyach from my experience developing the `TopicCommand` the reason for this specific type checking is the following.
The kafka-topics.sh tool (which use `TopicCommand`) has some options like `--create`, `--describe` and so on which doesn't have a related argument/value; they are a sort of "boolean" and you say that you want "to create" (true) but not "to describe" (false).
The way to have that is using `Arguments.storeTrue()` method.
For example : 
`
operationOptions.addArgument("--create")
                    .action(storeTrue())
                    .dest(CREATE)
                    .help("Create a new topic.");
`
If you don't use the `storeTrue()` method for such options without arg, an `ArgumentParserException` exception is raised.
Said that, when the parsing is executed, in the namespace attributes list you will have "create" entry with value true and "describe" entry with value false (even if you have't passed `--describe` on the command line).
Having just the line : 

`this.ns.get(option) != null`

means that for example in my `TopicCommand` case, for both "create" and "describe" this return true because they are in the attributes list (but "create" is true while "describe" is false).
To the command class it looks like the user has specified both `--create` and `--describe` which is not possible ... so an error is raised.
For this reason for `Boolean` values I need to check the value entry and not just checking that the entry is or not in the attributes list.

Reply:
Okay, it is clear now, thank you 👍 
However, I do not see you, using this method in the code. Instead, you just do e.g. 
`String bootstrapServer = opts.bootstrapServer();`
and then 
`if (bootstrapServer != null) {`
So you need a value in case it is not `null`, probably, that's why the `has` method won't be very usable. 
I am not against it, it adds no complexity, I am just not able to find a use case for it, except checking for flags, which can be checked by client code without any type checking.

Reply:
If you take a look at my WIP PR (https://github.com/apache/kafka/pull/3514) about the `TopicCommand` you can see that there are some other `CommandLineUtils` methods for checking arguments. The `has` method is used there and it provides a more concise way to check option existence. If you take a look at what is used in the current `TopicCommand` in Scala, the `has` method is provided by the `OptionSet` class. I'd like to give quite the same developer experience but in Java so the decision to add this method to the `CommandOptions` base class.